src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -651,42 +649,48 @@ public static void renderLitItem(RenderItem ri, IBakedModel model, int color, It
 
         allquads.addAll(model.getQuads(null, null, 0));
 
+        if (allquads.isEmpty()) return;
+
         // Current list of consecutive quads with the same lighting
         List<BakedQuad> segment = new ArrayList<>();
 
         // Lighting of the current segment
         int segmentBlockLight = -1;
         int segmentSkyLight = -1;
+        // Coloring of the current segment
         int segmentColorMultiplier = color;
-        
+        // If the current segment contains lighting data
+        boolean hasLighting = false;
+
         // Tint index cache to avoid unnecessary IItemColor lookups
         int prevTintIndex = -1;
 
         for (int i = 0; i < allquads.size(); i++) 
         {
-            BakedQuad q = allquads.get(i);
+            BakedQuad quad = allquads.get(i);
 
             // Lighting of the current quad
-            int bl = 0;
-            int sl = 0;
+            int blockLight = 0;
+            int skyLight = 0;
 
             // Fail-fast on ITEM, as it cannot have light data
-            if (q.getFormat() != DefaultVertexFormats.ITEM) 
+            if (quad.getFormat() != DefaultVertexFormats.ITEM && quad.getFormat().hasUvOffset(1))
```

Comment:
What is hasUvOffset for here?

Reply:
`hasUvOffset(1)` is a fast check for if lightmap data is expected, to avoid calling `pipe` unnecessarily.
The vertex format keeps count of how many uv elements it has (which is also why the consumer needs to have both).